model/src/application/callback/callback_data.rs

```diff
@@ -24,4 +25,47 @@ pub struct CallbackData {
     pub flags: Option<MessageFlags>,
     #[serde(skip_serializing_if = "Option::is_none")]
     pub tts: Option<bool>,
+    #[serde(default, skip_serializing_if = "Vec::is_empty")]
+    pub components: Vec<Component>,
+}
+
+impl CallbackData {
+    #[inline]
+    pub fn builder() -> Builder {
+        Builder::new()
+    }
+}
+
+#[derive(Debug)]
+pub struct Builder {
+    pub allowed_mentions: Option<AllowedMentions>,
+    pub content: Option<String>,
+    pub embeds: Vec<Embed>,
+    pub flags: Option<MessageFlags>,
+    pub tts: Option<bool>,
+}
```

Comment:
Builders should hold the resultant data structure in a tuple struct and modify it. They should also be named after the struct they are building.
```suggestion
#[derive(Debug)]
pub struct CallbackDataBuilder(CallbackData);
```